### PR TITLE
[ROS2] increase history_depth for subscrition

### PR DESCRIPTION
**Pull request by mitsudome-r**
This resolves issue : https://github.com/ros/diagnostics/issues/167.

This makes the following modfiication:
* The new parameter `history_depth` for diagnostic_aggregator node to enable users to set history depth size(i.e. queue size) for subscription.
* Change default depth size to 1000 from 1.

Signed-off-by: mitsudome-r <ryohsuke.mitsudome@tier4.jp>

### CODE REVIEW

**Comment by Karsten1987** [comment]
This looks good to me. Please target the PR against the ros2 branch. I'll take care of a backport to existing distributions and cut a release.

**Comment by mitsudome-r** [comment]
@Karsten1987 
Thank you for the review.
I changed the target to ros2-devel. I also rebased to the branch as well.

**Comment by tgreier** [comment]
@Karsten1987 Hi Karsten. We need this PR for Foxy. But, it looks like this PR has stalled. What needs to be done to progress it?

File: diagnostic_aggregator/src/aggregator.cpp [hunk 1]
```diff
@@ -96,9 +99,9 @@ Aggregator::Aggregator()
   other_analyzer_ = std::make_unique<OtherAnalyzer>(other_as_errors);
   other_analyzer_->init(base_path_);  // This always returns true
 
-  diag_sub_ = n_->create_subscription<DiagnosticArray>(
-    "/diagnostics", rclcpp::SystemDefaultsQoS(), std::bind(&Aggregator::diagCallback, this, _1));
-  agg_pub_ = n_->create_publisher<DiagnosticArray>("/diagnostics_agg", 1);
+  diag_sub_ = n_->create_subscription<diagnostic_msgs::msg::DiagnosticArray>(
+    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_), std::bind(&Aggregator::diagCallback, this, _1));
+  agg_pub_ = n_->create_publisher<diagnostic_msgs::msg::DiagnosticArray>("/diagnostics_agg", 1);
```
Comment (Karsten1987) [inline diagnostic_aggregator/src/aggregator.cpp]: To highlight the actual changes:

```suggestion
  diag_sub_ = n_->create_subscription<DiagnosticArray>(
    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_),
    std::bind(&Aggregator::diagCallback, this, _1));
  agg_pub_ = n_->create_publisher<DiagnosticArray>("/diagnostics_agg", 1);
```

**Review by Karsten1987 — APPROVED** [review]
Just a little nitpick, otherwise ~all~ almost good, except the linters.

**Comment by Karsten1987** [comment]
@tgreier The changes reflected as is in this PR can't be backported to Foxy due to ABI incompatibility.
One way I would be okay with is to "simply" hardcode the QoS to a `keep_last(1000)` which should solve the problem described in #167 and achieve same behavior as ROS1.
Opinions?

**Review by Karsten1987 — CHANGES_REQUESTED** [review]
actually, I've just seen that uncrustify and cpplint aren't happy. Mind addressing them?

You'll find the details in the GH actions, e.g.:
```
  [18.595s] 12: --- src/aggregator.cpp
  [18.595s] 12: +++ src/aggregator.cpp.uncrustify
  [18.595s] 12: @@ -103 +103,2 @@
  [18.595s] 12: -    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_), std::bind(&Aggregator::diagCallback, this, _1));
  [18.595s] 12: +    "/diagnostics", rclcpp::SystemDefaultsQoS().keep_last(history_depth_),
  [18.595s] 12: +    std::bind(&Aggregator::diagCallback, this, _1));
```

**Review by Karsten1987 — APPROVED** [review]

**Pull request merged**